### PR TITLE
feat: modal title becomes optional and hides header if undefined

### PR DESCRIPTION
**Pull request by JeffMarvin**
### Added

- Modal: `title` is now an optional prop, and will now hide the Header if the title is not provided 

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Modal/Modal.tsx [hunk 1]
```diff
@@ -69,12 +71,13 @@ export function Modal({
               stiffness: 300,
             }}
           >
-            <Header
-              title={title}
-              dismissible={dismissible}
-              onRequestClose={onRequestClose}
-            />
-
+            {showTitle && (
```
Comment (seanhealy) [inline packages/components/src/Modal/Modal.tsx]: Just thinking it might be nice to do

```suggestion
            {title != undefined && (
```

here.

Comment (JeffMarvin) [reply packages/components/src/Modal/Modal.tsx]: Let's do it!

**Comment by JeffMarvin** [comment]
@seanhealy PTAL - took a new approach and added tests!

File: packages/components/src/Modal/Modal.tsx [hunk 2]
```diff
@@ -12,13 +12,14 @@ interface ModalProps {
...
    */
-  readonly title: string;
+  readonly title?: string;
   readonly open?: boolean;
   readonly size?: keyof typeof sizes;
   /**
    * @default true
    */
   readonly dismissible?: boolean;
+  readonly showTitle?: boolean;
```
Comment (seanhealy) [inline packages/components/src/Modal/Modal.tsx]: It looks like this is no longer needed.

Comment (JeffMarvin) [reply packages/components/src/Modal/Modal.tsx]: Oh geez... yup, removed.

**Pull request merged**